substrate/frame/support/test/tests/construct_runtime.rs

```diff
@@ -256,24 +281,54 @@ pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
 pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<u32, RuntimeCall, Signature, ()>;
 pub type Block = generic::Block<Header, UncheckedExtrinsic>;
 
-frame_support::construct_runtime!(
-	pub struct Runtime
-	{
-		System: frame_system::{Pallet, Call, Event<T>, Origin<T>} = 30,
-		Module1_1: module1::<Instance1>::{Pallet, Call, Storage, Event<T>, Origin<T>},
-		Module2: module2::{Pallet, Call, Storage, Event<T>, Origin},
-		Module1_2: module1::<Instance2>::{Pallet, Call, Storage, Event<T>, Origin<T>},
-		NestedModule3: nested::module3::{Pallet, Call, Config<T>, Storage, Event<T>, Origin},
-		Module3: self::module3::{Pallet, Call, Config<T>, Storage, Event<T>, Origin<T>},
-		Module1_3: module1::<Instance3>::{Pallet, Storage, Event<T> } = 6,
-		Module1_4: module1::<Instance4>::{Pallet, Call, Event<T> } = 3,
-		Module1_5: module1::<Instance5>::{Pallet, Event<T>},
-		Module1_6: module1::<Instance6>::{Pallet, Call, Storage, Event<T>, Origin<T>} = 1,
-		Module1_7: module1::<Instance7>::{Pallet, Call, Storage, Event<T>, Origin<T>},
-		Module1_8: module1::<Instance8>::{Pallet, Call, Storage, Event<T>, Origin<T>} = 12,
-		Module1_9: module1::<Instance9>::{Pallet, Call, Storage, Event<T>, Origin<T>},
+#[frame_support::construct_runtime_v2]
+mod runtime {
+	#[frame::runtime]
+	pub struct Runtime;
+
+	#[frame::pallets]
+	#[frame::derive(
+		RuntimeCall,
+		RuntimeEvent,
+		RuntimeError,
+		RuntimeOrigin,
+		RuntimeFreezeReason,
+		RuntimeHoldReason,
+		RuntimeSlashReason,
+		RuntimeLockId
+	)]
+	pub struct Pallets {
+		#[frame::pallet_index(30)]
+		System: frame_system + Pallet + Call + Event<T> + Origin<T>,
+		#[frame::pallet_index(31)]
+		Module1_1: module1<Instance1>,
+		#[frame::pallet_index(32)]
+		Module2: module2,
+		#[frame::pallet_index(33)]
+		Module1_2: module1<Instance2>,
+		#[frame::pallet_index(34)]
+		NestedModule3: nested::module3,
+		#[frame::pallet_index(35)]
+		#[frame::disable_unsigned]
```

Comment:
yeah I stand by my comment about pallet indices, since _every_ item in the `Pallets` struct needs one, vs things like `disable_unsigned` which can be more verbose because they are not on every single item 👍🏻